### PR TITLE
Basic support for open graph meta tags for PDP

### PR DESCRIPTION
**Pull request by nnande**

### CODE REVIEW

File: core/app/helpers/spree/base_helper.rb [hunk 1]
```diff
@@ -57,8 +57,29 @@ def logo(image_path = nil, options = {})
...
 
+    def object
+      instance_variable_get('@' + controller_name.singularize)
+    end
+
+    def og_meta_data
+      og_meta = {}
+      if object.is_a? Spree::Product
+        image                             = default_image_for_product_or_variant(object)
+        og_meta['og:image']               = main_app.url_for(image.attachment) if image&.attachment
```
Comment (nnande) [inline core/app/helpers/spree/base_helper.rb:68]: not sure if `main_app` or `spree` should be used but this seems to work

Comment (nnande) [reply core/app/helpers/spree/base_helper.rb:68]: https://github.com/spree/spree/blob/224fadedf68f7670be913b41b5a1fb33407da095/guides/src/content/release_notes/1_0_0.md#referencing-spree-routes

File: core/app/helpers/spree/base_helper.rb [hunk 2]
```diff
@@ -57,8 +57,29 @@ def logo(image_path = nil, options = {})
...
+
+        og_meta['og:url']                 = spree.url_for(object) if frontend_available? # url_for product needed
+        og_meta['og:type']                = object.class.name.demodulize.downcase
+        og_meta['og:title']               = object.name
+        og_meta['og:description']         = object.description
+
+        og_meta['product:price:amount']   = object.master.default_price.amount
+        og_meta['product:price:currency'] = object.master.default_price.currency
```
Comment (damianlegawiec) [inline core/app/helpers/spree/base_helper.rb:76]:
```suggestion
        if (price = object.price_in(current_currency)
          og_meta['product:price:amount']   = price.amount
          og_meta['product:price:currency'] = current_currency
        end
```

Comment (nnande) [reply core/app/helpers/spree/base_helper.rb]: @damianlegawiec followed your suggestion, can be reviewed once again now

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**